apps/backend/src/mutations/TechniqueMutations.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  assignInstrumentsToTechniqueValidationSchema,
+  createTechniqueValidationSchema,
+  deleteTechniqueValidationSchema,
+  removeInstrumentsFromTechniqueValidationSchema,
+  updateTechniqueValidationSchema,
+} from '@user-office-software/duo-validation';
+import { inject, injectable } from 'tsyringe';
+
+import { UserAuthorization } from '../auth/UserAuthorization';
+import { Tokens } from '../config/Tokens';
+import { Authorized, EventBus, ValidateArgs } from '../decorators';
+import { Event } from '../events/event.enum';
+import { Rejection, rejection } from '../models/Rejection';
+import { Roles } from '../models/Role';
+import { Technique } from '../models/Technique';
+import { UserWithRole } from '../models/User';
+import { AssignInstrumentsToTechniqueArgs } from '../resolvers/mutations/AssignInstrumentsToTechnique';
+import { CreateTechniqueArgs } from '../resolvers/mutations/CreateTechniqueMutation';
+import { RemoveInstrumentFromTechniqueArgs } from '../resolvers/mutations/RemoveInstrumentFromTechnique';
+import { UpdateTechniqueArgs } from '../resolvers/mutations/UpdateTechniqueMutations';
+import { TechniqueDataSource } from './../datasources/TechniqueDataSource';
+
+@injectable()
+export default class TechniqueMutations {
+  constructor(
+    @inject(Tokens.TechniqueDataSource)
+    private dataSource: TechniqueDataSource,
+    @inject(Tokens.UserAuthorization) private userAuth: UserAuthorization
+  ) {}
+
+  @EventBus(Event.TECHNIQUE_CREATED)
+  @ValidateArgs(createTechniqueValidationSchema)
+  @Authorized([Roles.USER_OFFICER])
+  async create(
+    agent: UserWithRole | null,
+    args: CreateTechniqueArgs
+  ): Promise<Technique | Rejection> {
+    const technique = await this.dataSource.create(args).catch((error) => {
+      return rejection(
+        'Could not create technique',
+        { agent, shortCode: args.shortCode },
+        error
+      );
```

Comment:
I am not seeing the underlining function propagating any error message error will most likely be always not set.

Reply:
I missed this, thanks. Now the datasource throws errors with messages when something unexpected happens. Also made sure all mutations/queries catch the error and return a rejection including the reason and any relevant args.